File: android-app/src/main/java/com/medziku/plantwatering/data/parts/TimerImpl.java
package com.medziku.plantwatering.data.parts;

import android.os.Parcel;
import android.os.Parcelable;

import com.medziku.plantwatering.station.parts.Timer;


public class TimerImpl extends Timer implements Parcelable {

    public TimerImpl(String name, String time) {
        super(name, time);
    }

    private TimerImpl(Parcel in) {
        super(in.readString(), in.readString());
    }

    public static final Creator<TimerImpl> CREATOR = new Creator<TimerImpl>() {
        @Override
        public TimerImpl createFromParcel(Parcel in) {
            return new TimerImpl(in);
        }

        @Override
        public TimerImpl[] newArray(int size) {
            return new TimerImpl[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeString(getTime());
    }
}

File: android-app/src/main/kotlin/com/medziku/plantwatering/data/parts/TimerImpl.kt
package com.medziku.plantwatering.data.parts

import android.os.Parcel
import android.os.Parcelable

import com.medziku.plantwatering.station.parts.Timer


class TimerImpl : Timer, Parcelable {

    constructor(name: String, time: String) : super(name, time)

    private constructor(`in`: Parcel) : super(`in`.readString(), `in`.readString())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeString(time)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<TimerImpl> = object : Parcelable.Creator<TimerImpl> {
            override fun createFromParcel(`in`: Parcel): TimerImpl {
                return TimerImpl(`in`)
            }

            override fun newArray(size: Int): Array<TimerImpl?> {
                return arrayOfNulls(size)
            }
        }
    }
}
